docs/advanced_topics/accessibility_considerations.rst

```diff
@@ -0,0 +1,160 @@
+
+TableBlock
+----------
+
+The :doc:`/reference/contrib/table_block` default implementation makes it too easy for end users to miss they need either row or column headers (`#5989 <https://github.com/wagtail/wagtail/issues/5989>`_). Make sure to always have either row headers or column headers set.
+Also treat its Caption field as mandatory, so screen reader users navigating the site’s tables know where they are.
+
+----
+
+.. _accessibility_in_templates:
+
+Accessibility in templates
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here are common gotchas to be aware of to make the site’s templates as accessible as possible,
+
+Alt text in templates
+---------------------
+
+See the :ref:`content modelling <content_modeling>` section above. Additionally, make sure to :ref:`customise images’ alt text <image_tag_alt>`, either setting it to the relevant field, or to an empty string for decorative images, or images where the alt text would be a repeat of other content.
+Even when your images have alt text coming directly from the image model, you still need to decide whether there should be alt text for the particular context the image is used in. For example, avoid alt text in listings where the alt text just repeats the listing items’ title.
+
+Empty heading tags
+------------------
+
+In both rich text and custom StreamField blocks, it’s sometimes easy for editors to create a heading block but not add any content to it. If this is a problem for your site,
+
+- Add validation rules to those fields, making sure the page can’t be saved with the empty headings.
+- Consider adding similar validation rules for rich text fields.
+
+Additionally, you can hide empty heading blocks with CSS:
+
+.. code-block:: css
+
+    h1:empty, h2:empty, h3:empty, h4:empty, h5:empty, h6:empty {
+        display: none;
+    }
+
+In the future, :ref:`rich text rewrite handlers <rich_text_rewrite_handlers>` should also support this being done server-side (`#4223 <https://github.com/wagtail/wagtail/issues/4223>`_).
+
+Forms
+-----
+
+When using the ``wagtailforms`` :ref:`form_builder`, don’t stop at Django’s default forms rendering:
+
+- Avoid ``as_table`` and ``as_ul``, which make forms harder to navigate for screen reader users.
+- Take the time to customize the default form rendering so required fields are clearly labeled as such, so fields use the correct autocomplete and autofill attributes, etc.
+- Follow `official W3C guidance on accessible forms development <https://www.w3.org/WAI/tutorials/forms/>`_.
+
+----
+
+.. _authoring_accessible_content:
+
+Authoring accessible content
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here are things you can do to help content authors take accessibility into consideration.
```

Comment:
Thank you, I like it, will change to `create` rather than `compose`.

```suggestion
Here are things you can do to help authors create accessible content.
```